structuredheadings/plugin.js

```diff
@@ -464,6 +466,113 @@
             setStyle(editor, node, editor.config.autonumberCurrentStyle);
           }
         }
+      },
+
+      /*
+       * convertList
+       */
+      convertList: {
+        exec: function (editor) {
+
+          var selectArea = function (startElement, endElement, encompass) {
+            encompass = encompass ? encompass : false;
+            var range = editor.createRange();
+            if (encompass) {
+              range.setStartBefore(startElement, 0);
+              range.setEndAfter(endElement, 0);
+            } else {
+              range.setStart(startElement, 0);
+              range.setEnd(endElement, 0);
+            }
+            editor.getSelection().selectRanges([ range ]);
+          };
+
+          /* The Worker */
+          var processList = function (listElement, level, deep) {
+            level = level ? level : 0;
+            deep = deep ? deep : false;
+
+            listElement.forEach(function (node) {
+              if (node.type === CKEDITOR.NODE_ELEMENT && node.is({li: 1})) {
+
+                selectArea(node, node);
+
+                // Apply the element style to the selection based on current level
+                //eslint-disable-next-line new-cap
+                editor.applyStyle(new CKEDITOR.style({
+                  element: editor.config.numberedElements[level]
+                }));
+
+                // Apply numbering also
+                editor.execCommand("autoNumberHeading");
+
+                return true;
+
+              /* if another list is found, and deep is enabled, recursively call the worker
+               * and incremement the level, deep will convert the entire list
+               */
+              } else if (node.type === CKEDITOR.NODE_ELEMENT && node.is({ul: 1, ol: 1}) && deep) {
+                processList(node, level + 1);
+                return false;
+              } else {
+                return false;
+              }
+            }, null, true);
+          };
+
+          //get the current list root element from the selection
+          var listRoot = editor.elementPath().contains(["ol", "ul"], false, true);
+          var deep = false; //don't deep process
```

Comment:
I have it in there in case Wes changes his mind how he wants the conversion to work. setting deep true will enable recursion, and convert the entire list from head to toe instead of just the "first level"